src/Pack/Core/Logging.idr

```diff
@@ -55,6 +99,29 @@ export %inline
 prompt : HasIO io => (lvl : LogLevel) -> (msg : String) -> io String
 prompt lvl msg = log (MkLogRef lvl) lvl msg >> map trim getLine
 
+||| Requests confirmation from the user with the specified message and
+||| parses a reply from stdin as `Yes` or `No`.
+|||
+||| This uses the given log level but makes sure the message is always
+||| printed no matter the current log level preferences.
+export %inline
+confirm : HasIO io => (lvl : LogLevel) -> (msg : String) -> io ConfirmResult
+confirm lvl msg =
+  map parseConfirmResult $ prompt lvl $ confirmMessage msg
+
+||| Requests confirmation from the user with the specified message and
+||| If the reply is not `Yes`, aborts with `SafetyAbort`.
+|||
+||| This uses the given log level but makes sure the message is always
+||| printed no matter the current log level preferences.
+export %inline
+confirmOrAbort :
+     HasIO io
+  => (lvl : LogLevel)
```

Comment:
Please don't you double arrows in multi-line type signatures. Use `{auto ...}` instead, for better readability.